cms/djangoapps/contentstore/course_group_config.py

```diff
@@ -84,7 +85,7 @@ def assign_group_ids(self):
         """
         Assign ids for the group_configuration's groups.
         """
-        used_ids = [g.id for p in self.course.user_partitions for g in p.groups]
+        used_ids = [g.id for p in get_course_user_partitions(self.course) for g in p.groups]
```

Comment:
@nasthagiri I chose to not replace all usages of self.course.user_partitions, as most calls really only want the partitions stored in course.user_partitions (for example, when looking specifically at cohort, random, or verification partitions).

When I did need all partitions, I used this helper method get_course_user_partitions unless I didn't have access to the course, in which case I went through the runtime service.

This particular call needs all partitions because it is what the Studio component visibility dialog uses (though this file is actually one that would be removed before merging the PR, as it is only present for local testing).